example_allocators/compartment_alloc/compartment_alloc.c

```diff
@@ -0,0 +1,91 @@
+#include <cheriintrin.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/mman.h>
+#include "compartment_alloc.h"
+#include "utils.h"
+
+heap_compartment compartments[maxCompartments];
```

Comment:
Just double checking this does get initialized automatically because it's a global, and we don't have to explicitly `malloc` it.

Reply:
Yep: https://stackoverflow.com/a/16015741